### ISSUE TITLE
Truncating and Limiting the Value's class toString() to 36 characters 

### ISSUE THREAD

**Issue by gauravpurohit06** [googleapis/java-spanner#2654]
In java class [Value](https://github.com/googleapis/java-spanner/blob/main/google-cloud-spanner/src/main/java/com/google/cloud/spanner/Value.java) for 3 data types which includes `String`, `JSON`,  `PGJSONB` we are truncating the response of method `valueToString` to 36 characters.  Method `valueToString` is called internally by `toString()` method. 

The `toString()` method is designed to return a string representation of an object, and it's expected to provide a complete and meaningful representation of the object's state. Truncating it may lead to loss of information and make debugging and logging more difficult.

**Comment by gauravpurohit06** [comment]
**Context**
While working on Dataflow Integration Tests, I am converting mutation object to map and then using toString() to get a valid string but the string is truncated as it's internally using the above method.

**Comment by olavloite** [comment]
You should call [`Value#getAsString()`](https://github.com/googleapis/java-spanner/blob/efe95b0f45df184cfcdcacfcd5b556111fded7c4/google-cloud-spanner/src/main/java/com/google/cloud/spanner/Value.java#L741) for that

**Comment by gauravpurohit06** [comment]
@olavloite, yes we can use it but the thing is we are not calling it directly on any particular Value instance. We are using `Mutation.toString()`  which internally calls `toString` method for all the Value objects.

**Comment by olavloite** [comment]
For that, we should add a method to `Mutation` (e.g. `Mutation#getStringRepresentation()`) that calls `Value#getAsString()`. Changing the `toString()` implementation of either `Value` or `Mutation` could be considered a breaking change, as it could risk that someone who is currently logging the outcome of those methods would all of a sudden see a log size explosion.

**Comment by gauravpurohit06** [comment]
Thanks @olavloite, it makes sense. Client Library owners to  take the call, if it's helpful to add.
